hashring/iter_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
```

Comment:
This file (previously rbtree/iter_test.go) did not change much. I mostly rewritten it to use for loops.